sway/desktop/output.c

```diff
@@ -651,4 +673,72 @@ void handle_new_output(struct wl_listener *listener, void *data) {
 	}
 
 	transaction_commit_dirty();
+
+	update_output_manager_config(server);
+}
+
+void handle_output_manager_apply(struct wl_listener *listener, void *data) {
+	struct sway_server *server =
+		wl_container_of(listener, server, output_manager_apply);
+	struct wlr_output_configuration_v1 *config = data;
+
+	struct wlr_output_configuration_head_v1 *config_head;
+	// First disable outputs we need to disable
+	wl_list_for_each(config_head, &config->heads, link) {
+		struct wlr_output *wlr_output = config_head->state.output;
+		struct sway_output *output = wlr_output->data;
+		if (!output->enabled || config_head->state.enabled) {
+			continue;
+		}
+		struct output_config *oc = new_output_config("temp");
+		oc->enabled = false;
+		// There is no explicit way to set DPMS through the protocol.
+		// Users probably expect that disabling a monitor will turn it off.
+		oc->dpms_state = DPMS_OFF;
+
+		apply_output_config(oc, output);
+		free_output_config(oc);
+	}
+
+	// Then enable outputs that need to
+	wl_list_for_each(config_head, &config->heads, link) {
+		struct wlr_output *wlr_output = config_head->state.output;
+		struct sway_output *output = wlr_output->data;
+		if (!config_head->state.enabled) {
+			continue;
+		}
+		struct output_config *oc = new_output_config("temp");
+		oc->enabled = true;
+		if (config_head->state.mode != NULL) {
+			struct wlr_output_mode *mode = config_head->state.mode;
+			oc->width = mode->width;
+			oc->height = mode->height;
+			oc->refresh_rate = mode->refresh;
+		} else {
+			oc->width = config_head->state.custom_mode.width;
+			oc->height = config_head->state.custom_mode.height;
+			oc->refresh_rate = config_head->state.custom_mode.refresh;
+		}
+		oc->x = config_head->state.x;
+		oc->y = config_head->state.y;
+		oc->transform = config_head->state.transform;
+		oc->scale = config_head->state.scale;
+
+		apply_output_config(oc, output);
+		free_output_config(oc);
+	}
+
+	// Handle failure somehow?
```

Comment:
Could make `apply_output_config` return a bool, if it doesn't do that yet

Reply:
It already does, my bad